src/System.Globalization.Extensions/src/System/Globalization/Extensions.cs

```diff
@@ -26,72 +26,8 @@ public static StringComparer GetStringComparer(this CompareInfo compareInfo, Com
                 return StringComparer.OrdinalIgnoreCase;
             }
 
-            if ((options & CultureAwareComparer.ValidCompareMaskOffFlags) != 0)
-            {
-                throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
-            }
-
```

Comment:
I see this in the .csproj:
https://github.com/Anipik/corefx/blob/c9a7d5c3e4fb4a1ef5897cb0c03440044922a986/src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj#L20-L21
and this should fail when built for netfx as the below overload doesn't exist there, but I see that build passed CI, and it looks like that's because we don't build this library for netfx:
https://github.com/Anipik/corefx/blob/c9a7d5c3e4fb4a1ef5897cb0c03440044922a986/src/System.Globalization.Extensions/src/Configurations.props#L5-L7
@tarekgh, is that just cruft that can be cleaned up?

Reply:
Yes we don't build netfx package for this library anymore. mainly most of the functionality are already exposed from somewhere else. 

@Anipik we just need to delete all the cs files there and clean up the csproj files. 


 

---
In reply to: [167985111](https://github.com/dotnet/corefx/pull/27099#discussion_r167985111) [](ancestors = 167985111)

Reply:
@tarekgh can you please tell which files need to be removed ?

Reply:
We need to delete the files

https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/src/System/Globalization/Extensions.cs

https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/src/System/StringNormalizationExtensions.netfx.cs

https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/src/ApiCompatBaseline.netfx.txt

And then clean up the file 
https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/src/System.Globalization.Extensions.csproj

to remove any support for netfx there. 

then we need to update the implementation in the file 

https://github.com/dotnet/corefx/blob/a3e4e241c7b7c33750258082969287b9cf45d757/src/System.Runtime.Extensions/src/System/Globalization/Extensions.cs

to remove the CultureAwareComparer and modify the extension method there.

cc @weshaggard